File: leetcode/BinarySearch/leetcode-69/69.go
package binarysearch

func mySqrt(x int) int {
	if x < 2 {
		return x
	}

	left, right := 0, x
	for left < right {
		mid := (right-left)/2 + left
		if x/mid >= mid {
			left = mid + 1
		} else {
			right = mid
		}
	}
	return right - 1
}

File: leetcode/BinarySearch/leetcode-69/69.java
class Solution {
    public int mySqrt(int x) {
        if (x < 2) {
            return x;
        }
        int left = 0, right = x;
        while (left < right) {
            int mid = (right - left) / 2 + left;
            if (x / mid >= mid) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return right - 1;
    }
}